test/integration/src/test/java/io/pravega/test/integration/ReaderGroupNotificationTest.java

```diff
@@ -189,19 +213,128 @@ public void testSegmentNotifications() throws Exception {
+        @Cleanup
+        ConnectionFactory connectionFactory = new ConnectionFactoryImpl(ClientConfig.builder()
+                .controllerURI(URI.create("tcp://localhost"))
+                .build());
+        @Cleanup
+        ClientFactoryImpl clientFactory = new ClientFactoryImpl(SCOPE, controller, connectionFactory);
+        EventStreamWriter<String> writer1 = clientFactory.createEventWriter(streamName, new JavaSerializer<>(),
+                EventWriterConfig.builder().build());
+        List<String> routingKeys = new ArrayList<>();
+        routingKeys.add("0");
+        routingKeys.add("1");
+        routingKeys.add("2");
+        routingKeys.add("3");
+        writer1.writeEvent("0", "data").get();
+
+        // scale
+        Stream stream = new StreamImpl(SCOPE, streamName);
+        writer1.writeEvent("0", "data").get();
+
+        @Cleanup
+        ReaderGroupManager groupManager = new ReaderGroupManagerImpl(SCOPE, controller, clientFactory,
+                connectionFactory);
+        ReaderGroupConfig readerGroupConfig = ReaderGroupConfig
+                .builder().disableAutomaticCheckpoints().stream(Stream.of(SCOPE, streamName)).groupRefreshTimeMillis(0).build();
+        groupManager.createReaderGroup("reader", readerGroupConfig);
+        @Cleanup
+        ReaderGroup readerGroup = groupManager.getReaderGroup("reader");
+
+        long start = System.currentTimeMillis();
+        log.info("Segments found as: {}", controller.getCurrentSegments(SCOPE, streamName).get());
+        CompletableFuture.runAsync(() -> {
+            while (System.currentTimeMillis() - start < Duration.ofMinutes(1).toMillis()) {
+                try {
+                    writer1.writeEvent(routingKeys.get(random.nextInt(3)), "data").get();
+                    Thread.sleep(0);
+                } catch (Throwable e) {
+                    log.error("test exception: ", e);
+                    throw new IllegalStateException(e);
+                }
+            }
+        }, threadPoolExecutor)
+                .exceptionally(e -> {
+                    log.error("Failure: ", e);
+                    throw new IllegalStateException(e);
+                });
+        try {
+            Thread.sleep(2000);
+            scale(controller, SCOPE, streamName, 1, 1, threadPoolExecutor);
+            Thread.sleep(2000);
+        } catch (InterruptedException e) {
+            log.error("Interrupted: ", e);
+        }
+        Retry.withExpBackoff(10, 10, 100, 10000)
+                .retryingOn(IllegalStateException.class)
+                .throwingOn(RuntimeException.class)
+                .runAsync(() -> controller.getCurrentSegments(SCOPE, streamName)
+                        .thenAccept(streamSegments -> {
+                            try {
+                                Thread.sleep(1000);
+                                EventStreamReader<String> reader1 = clientFactory.createReader("readerId", "reader", new JavaSerializer<>(),
```

Comment:
use `readerGroup.getName` instead of hardcoding the name.

Reply:
Also put `readerId` in a constant to avoid copy-pasting everywhere.

Reply:
Yes Andre